v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compiler/index.ts

```diff
@@ -77,17 +77,22 @@ export async function downloadConfiguredCompilers(
   }
 }
 
-export async function getCompiler(version: string): Promise<Compiler> {
-  const platform = CompilerDownloaderImplementation.getCompilerPlatform();
-  const compilerDownloader = new CompilerDownloaderImplementation(
-    platform,
-    await getGlobalCompilersCacheDir(),
-  );
+export async function getCompiler(
+  version: string,
+  preferWasm: boolean,
```

Comment:
Can we make this an options object? Otherwise the calls to it look like this:

```ts
getCompiler(version, false)
```

instead of the much clearer:

```ts
getCompiler(version, { preferWasm: false })
```